functions/paymentCollector/src/index.ts

```diff
@@ -0,0 +1,88 @@
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import { marshall } from "@aws-sdk/util-dynamodb";
+import { DynamoDBDocumentClient, QueryCommand, UpdateCommand } from '@aws-sdk/lib-dynamodb';
+import type { Event } from "./event";
+
+const client = new DynamoDBClient({ region: "us-west-2" });
+const ddb = DynamoDBDocumentClient.from(client);
+
+export const handler = async (event: Event) => {
+  const body = event.body
+  const tableName = event.environment === "prod" ? "paymentJobs-prod" : "paymentJobs"
+  let message = "Unknow status"
+  let statusCode: number | undefined = 200
+
+  if (!await paymentExist(body.pk, body.sk, tableName)) {
+    throw new Error("Payment does not exisist!")
+  }
+
+  const input = {
+    "ExpressionAttributeNames": {
+      "#S": "status",
+      "#DT": "details"
+    },
+    "ExpressionAttributeValues": {
+      ":s": {
+        "S": "PAID"
+      },
+      ":dt": {
+        "M": marshall(body.details)
+      }
+    },
+    "Key": {
+      "pk": {
+        "S": body.pk
+      },
+      "st": {
+        "S": body.sk
+      }
+    },
+    "ReturnValues": "ALL_NEW",
+    "TableName": tableName,
+    "UpdateExpression": "SET #S = :s, #DT = :dt"
+  };
+
+  const command = new UpdateCommand(input);
+
+  try {
+    const result = await ddb.send(command)
+    statusCode = result.$metadata.httpStatusCode
+    message = "Rent collected successfully"
```

Comment:
I just got a thought about strings coming back from the server. What if instead of this we return the status code 200 plus an id or code, instead of a string as we could have the strings centralized within the UI and the APIs won't need to change if for whatever reason we want to update the message rendered when this is success.

Reply:
yes, also agree on this point, it seems there is a `APIGatewayProxyResult` I think it is a good idea to type the return of these function as `Promise<APIGatewayProxyResult> ` to keep them consistent. 

One verification I have not done yet, is to check if the statusCode returned by the lambda actually modify the status code of the http headers in the response.